### PR TITLE
fix: move algolia keys into code

### PR DESCRIPTION
**Pull request by uptownhr**

### CODE REVIEW

**Review by fullykubed — CHANGES_REQUESTED** [review]
These are build time arguments so should not be moved into code but rather added to the website_image_builder as build args. See https://panfactum.com/docs/edge/reference/infrastructure-modules/submodule/workflow/wf_dockerfile_build#input_args.

File: packages/reference/infrastructure/pf_website/vars.tf [hunk 1]
```diff
@@ -13,20 +13,4 @@ variable "vpa_enabled" {
...
-
-variable "algolia_app_id" {
-  description = "The Algolia App ID for the search index"
-  type        = string
-}
-
-variable "algolia_search_api_key" {
-  description = "The Algolia Search API key for the search index"
-  type        = string
-  sensitive   = true
```
Comment (fullykubed) [inline packages/reference/infrastructure/pf_website/vars.tf:26]: This should not be marked sensitive as you said it isn't sensitive.

File: packages/website/Containerfile [hunk 2]
```diff
@@ -41,8 +41,10 @@ WORKDIR /code/packages/website
 COPY packages/website/ .
```
Comment (fullykubed) [inline packages/website/Containerfile:41]: This needs to be updated to define the algolia build arguments.

**Review by fullykubed — CHANGES_REQUESTED** [review]

File: packages/reference/infrastructure/pf_website/vars.tf (hunk 1 above)
Comment (uptownhr) [reply packages/reference/infrastructure/pf_website/vars.tf:26]: being removed as the deployment env has no effect on a built next build

Comment (fullykubed) [reply packages/reference/infrastructure/pf_website/vars.tf:26]: No you should still keep them as input variables. We want to be able to change this without changing the code.

File: packages/website/src/lib/constants.ts [hunk 3]
```diff
@@ -32,14 +32,9 @@ export function isValidVersionSlug (maybeSlug: string | undefined): maybeSlug is
...
 export function algoliaEnv () {
-  const ALGOLIA_APP_ID = process.env.NEXT_PUBLIC_ALGOLIA_APP_ID
-  const ALGOLIA_SEARCH_API_KEY = process.env.NEXT_PUBLIC_ALGOLIA_SEARCH_API_KEY
-  const ALGOLIA_INDEX_NAME = process.env.NEXT_PUBLIC_ALGOLIA_INDEX_NAME
-
-  if (ALGOLIA_APP_ID === undefined || ALGOLIA_SEARCH_API_KEY === undefined || ALGOLIA_INDEX_NAME === undefined) {
-    console.log('algoliaEnv', ALGOLIA_APP_ID, ALGOLIA_SEARCH_API_KEY, ALGOLIA_INDEX_NAME)
-    throw new Error('Missing Algolia environment variables')
-  }
+  const ALGOLIA_APP_ID = process.env.NEXT_PUBLIC_ALGOLIA_APP_ID || 'VJ9GF38NJX'
```
Comment (fullykubed) [inline packages/website/src/lib/constants.ts:35]: We should not hardcode these like this. It will make identifying errors more difficult in the future.

**Review by fullykubed — CHANGES_REQUESTED** [review]

File: packages/reference/infrastructure/demo_cicd/vars.tf [hunk 4]
```diff
@@ -57,6 +57,12 @@ variable "algolia_api_key" {
   sensitive   = true
 }
 
+variable "algolia_search_api_key" {
+  description = "The Algolia Search API key for the search index that can be publicly exposed"
+  type        = string
+  sensitive   = true
```
Comment (fullykubed) [inline packages/reference/infrastructure/demo_cicd/vars.tf:63]: Not sensitive.

**Review by fullykubed — CHANGES_REQUESTED** [review]

**Review by fullykubed — APPROVED** [review]

**Pull request merged**